test/response/header.js

```diff
@@ -10,6 +12,30 @@ describe('res.header', () => {
     res.header.should.eql({ 'x-foo': 'bar' });
   });
 
+  it('should use res.getHeaders() accessor when available', () => {
+    const res = response();
+    res.res._headers = null;
+    res.res.getHeaders = () => ({ 'x-foo': 'baz' });
+    res.header.should.eql({ 'x-foo': 'baz' });
+  });
+
+  it('should return the response header object when no mocks are in use', done => {
```

Comment:
@fengmk2 something like this? I avoided using context helper to get the real `http.ServerResponse`